src/components/FilteredList/FilteredList.tsx

```diff
@@ -0,0 +1,144 @@
+import * as React from 'react'
+import getValidProps from '@helpscout/react-utils/dist/getValidProps'
+import propConnect from '../PropProvider/propConnect'
+import { classNames } from '../../utilities/classNames'
+import { namespaceComponent } from '../../utilities/component'
+import { noop } from '../../utilities/other'
+import { isString, isObject } from '../../utilities/is'
+import { COMPONENT_KEY } from './FilteredList.utils'
+
+import Text from '../Text'
+import Tooltip from '../Tooltip'
+
+import {
+  FilteredListUI,
+  BadgeItemUI,
+  BadgeUI,
+  ItemUI,
+  SeparatorUI,
+} from './styles/FilteredList.css.js'
+
+export interface Props {
+  className?: string
+  items: string[]
+  itemKey?: string
+  limit: number
+  inline?: boolean
+  renderItem?: (object) => React.ReactNode
+}
+
+export class FilteredList extends React.Component<Props> {
+  static defaultProps = {
+    items: [],
+    limit: 5,
+  }
+
+  shouldComponentUpdate(nextProps) {
+    const { items } = this.props
+
+    if (nextProps.items.length !== items.length) {
+      return true
+    }
+
+    const itemKey = items
```

Comment:
Ohh, this is a niffy way of doing this check!

Reply:
Since it was part of the customer list table, we had to ensure this component render only when needed. Arrays/objects in props are not helpful for PureComponent so we went with a custom shouldComponentUpdate